daliuge-translator/dlg/dropmake/web/lg_web.py

```diff
@@ -218,11 +218,14 @@ def pgtcwl_get():
 
         # get paths used while creating the CWL files
         root_path = pgt_path("")
-        cwl_path = pgt_path(cwl_filename)
         zip_path = pgt_path(zip_filename)
 
         # create the CWL workflow
-        create_workflow(pgtp.drops, root_path, cwl_path, zip_path);
+        zip_contents = create_workflow(pgtp.drops, cwl_filename);
+
+        # maybe eventually we don't need to write to file
+        with open(zip_path, 'wb') as f:
+            f.write(zip_contents)
```

Comment:
I decided against adding a "file" parameter to create_workflow() for the reason you point out here. It would be better to write the zip contents directly to the HTTP response, without going via a file. It seemed better to have the flexibility to call create_workflow() without requiring a file.